### ISSUE TITLE
I found a solution.

### ISSUE THREAD

**Issue by poppinace** [poppinace/indexnet_matting#31]
I found a solution.

NumPy produces the same "random" values for every worker thread and every epoch because `torch.utils.data.DataLoader` forks the entire process, including the state of the random number generator: https://github.com/pytorch/pytorch/issues/5059

The fix is to seed the RNG differently using `worker_init_fn`.

I get MSE 0.01286 and SAD 43.8 after just 23 epochs.

_Originally posted by @983 in https://github.com/poppinace/indexnet_matting/issues/17#issuecomment-818694167_
